src/Analyser/TypeSpecifier.php

```diff
@@ -681,14 +679,35 @@ public function specifyTypesInCondition(
 					throw new ShouldNotHappenException();
 				}
 
-				return array_reduce(
-					array_filter(
-						$expr->vars,
-						static fn (Expr $var) => $scope->issetCheck($var, static fn () => true),
-					),
-					fn (SpecifiedTypes $types, Expr $var) => $types->unionWith($this->specifyTypesInCondition($scope, $var, $context, $rootExpr)),
-					new SpecifiedTypes(),
-				);
+				$specifiedTypes = new SpecifiedTypes();
+				foreach ($expr->vars as $var) {
+					$isset = $scope->issetCheck($var, static fn () => true);
+
+					if ($isset !== true) {
+						continue;
+					}
+
+					$type = $scope->getType($var);
+					if (
+						!$var instanceof Expr\Variable ||
```

Comment:
I'm not sure what's special about `Variable` that it needs to be mentioned in this condition, it's usually a red flag.

Reply:
Its the first step to get this better as before. 

Isset on variable is a very common case. I will make this more general in followups

Reply:
Please make this correct now, it's the right time to do it. The logic is correct when there is no special handling of a specific expr types.